docs/contrib/workflow.md

```diff
@@ -0,0 +1,232 @@
+|-----------------------------------------------|-----------------------------|-----------------------------|------------------|
+| HA flows Part 1: Added API objects            | `develop`                   | `feature/ha_flow_api`       | Related to #1234 |
+| HA flows Part 2: Added messaging objects      | `feature/ha_flow_api`       | `feature/ha_flow_messaging` | Related to #1234 |
+| HA flows Part 3: Implemented create operation | `feature/ha_flow_messaging` | `feature/ha_flow_create`    | Related to #1234 |
+| HA flows Part 4: Implemented delete operation | `feature/ha_flow_create`    | `feature/ha_flow_delete`    | Closes #1234     |
+
+
+If your PR has more than 1000 lines of code - please try to split in into several PRs.
+It is not mandatory rule, but it is highly recommended. It is very hard to review
+1000+ lines of code.
+
+### Pull request labels
+Your PR must have labels. We have different label categories.
+1. PR type. One of three labels: `feature`, `bugfix`, `improvment`.
+   Pick one which describes your changes best.
+2. Area. `area/testing`, `area/docs`, etc. Describes area which you are changing.
+3. Migration. If your PR contains database migration, you must specify label `DB migration`.
+4. Review status. Label `On review` must be set if your changes are ready for review.
+   If work is still in progress - do not use this label.
+5. Test status. Labels `Ready for Testing` and `tested` were made for QAs to track test
+   status of a PR. These labels will be described in [paragraph](#pull-request-life-cycle)
+   bellow.
+6. Merge status. If your PR has green tests, 2 approvals and label `tested` - it can be marked
+   as `Ready to merge`. It means that this PR can be included into next Release any minute.
+7. Release status. If your RP was merged, a release manager will change `Ready to merge` label 
+   with `Next Release`. You can't set label `Next Release`. Only release menager can.
+
+### Pull request life cycle
+Complete process of PR life cycle looks like:
+
+1. A developer opens a PR with labels from [categories](#pull-request-labels) 1-3.
+2. When the PR is no longer a draft, the developer adds reviewers and puts label `On Review`.
+3. When the task is completed and the PR can be built and deployed, a developer adds 
+   `Ready for Testing` label. It would be a signal for testers to start working on the
+   PR verification. 
+4. After a tester verified the current state of code in the PR (it's not necessary 
+   to have the approved PR at this moment), the tester replaces `Ready for Testing`
+   label with `tested`.
+5. If the tester finds the task is not properly solved (contains bugs, not covers
+   all requirements to close the task or other concerns), the tester removes the
+   `Ready for Testing` label and returns it to the development with the comment
+   which describe what is required to be fixed.
+6. If a developer needs to change something in a tested PR, then after pushing changes, 
```

Comment:
We miss the step where autotests are merged to the feature branch (unless you prefer to merge them later to `develop`).
The easier to to follow algorithm is to merge tests into feature branch after it got `Ready to Merge` label.
As an option, we can say that tests can be merged at any moment when tester believes that the tests cover the feature and use the right APIs. But in this case the later maintenance and functional test runs against the feature/bugfix would become a developer's task.

Reply:
There are several problems with merging tests into feature branch
1. We are loosing test discussion history. Merged code will point on feature PR, but all tests discussion will be on test PR
2. We can loose test PR in release notes. Sometimes PRs breaks something in release. In this case to check what was included in release a release manages checks release notes. If a release manager will merge only PR witth feature and tests - the manager will loose a link to tests PR.

The general approach is:
1. release manager merge feature into develop
2. deletes feature branch
3. GitHub changes base branch of PR with tests to `develop` automatically.
4. Release manager merge tests into develop